src/main/java/uk/gov/ons/ctp/response/sample/ingest/CsvIngesterBusiness.java

```diff
@@ -85,47 +85,38 @@ public CsvIngesterBusiness() {
     columnPositionMappingStrategy.setColumnMapping(COLUMNS);
   }
 
-  public SampleSummary ingest(MultipartFile file)
+  public SampleSummary ingest(final SampleSummary sampleSummary, final MultipartFile file)
       throws Exception {
 
     CSVReader csvReader = new CSVReader(new InputStreamReader(file.getInputStream()), ':');
     String[] nextLine;
-    SampleSummary sampleSummary;
-    BusinessSurveySample businessSurveySample = new BusinessSurveySample();
     List<BusinessSampleUnit> samplingUnitList = new ArrayList<>();
-    Integer expectedCI;
+    int lineNumber = 0;
 
       while((nextLine = csvReader.readNext()) != null) {
-
-        BusinessSampleUnit businessSampleUnit = processLine(columnPositionMappingStrategy, nextLine);
-        Optional<String> namesOfInvalidColumns = validateLine(businessSampleUnit);
-        if (namesOfInvalidColumns.isPresent()) {
-          log.error("Problem parsing line {} due to {} - entire ingest aborted", Arrays.toString(nextLine),
-              namesOfInvalidColumns.get());
-          throw new CTPException(CTPException.Fault.VALIDATION_FAILED, String.format("Problem parsing line %s due to %s", Arrays.toString(nextLine),
-              namesOfInvalidColumns.get()));
+        lineNumber++;
+
+        try {
+          BusinessSampleUnit businessSampleUnit = processLine(columnPositionMappingStrategy, nextLine);
+          Optional<String> namesOfInvalidColumns = validateLine(businessSampleUnit);
+          if (namesOfInvalidColumns.isPresent()) {
+            log.error("Problem parsing line {} due to {} - entire ingest aborted", Arrays.toString(nextLine),
+                    namesOfInvalidColumns.get());
+            throw new CTPException(CTPException.Fault.VALIDATION_FAILED, String.format("Error in %s due to field(s) %s", Arrays.toString(nextLine),
+                    namesOfInvalidColumns.get()));
+          }
+          businessSampleUnit.setSampleUnitType("B");
+
+          samplingUnitList.add(businessSampleUnit);
+        } catch(CTPException e){
+          String oldMessage = e.getMessage();
```

Comment:
I personally wouldn't bother assigning this to a variable just to call it once in the line below. It also means we can give newMessage a more descriptive name like exceptionMessage or just message

Reply:
Fair enough.

Reply:
Maybe it would be just better to wrap the exception https://github.com/ONSdigital/rm-common-service/blob/master/src/main/java/uk/gov/ons/ctp/common/error/CTPException.java#L72